packages/gestalt/src/Spinner/VRSpinner.css

```diff
@@ -0,0 +1,111 @@
+.spinner {
+  --g-dot-size: calc(var(--g-size) / 3);
+  --g-dot-space: calc(var(--g-dot-size) * 1.75);
+
+  height: var(--g-size);
+  width: var(--g-size);
+}
+
+.lg {
+  --g-size: var(--comp-spinner-size-lg);
+}
+
+.md {
+  --g-size: var(--comp-spinner-size-md);
+}
+
+.sm {
+  --g-size: var(--comp-spinner-size-sm);
```

Comment:
what does --g-size mean? / where is it used?

Reply:
Oh I see, you're using it above in the `.spinner` context

Reply:
I think - to simplify this further -> ` --g-dot-size: calc(var(--g-size) / 3);`  is actually 3 variations of a token.

So in theory, what this is effectively making is almost 9 variations of the calculated set

```
--g-dot-size-sm
--g-dot-size-lg
--g-dot-size-md
...
```

# Option 1
So maybe this may not be right in one sense. Because that's a lot of tokens.

I don't think the token should have the size baked into it. Maybe the way to architect this is with singular tokens.

```
--comp-spinner-size-height
--comp-spinner-size-width
--comp-dot-size
--comp-dot-space
```

I think the ideal approach is we have sizing similar to dark mode - and we can override the tokens. We need a sizing provider, and it needs to handle sizes across everything. I'm not sure what our bandwith is to do that rn.

# Option 2

You might be able to move this value definition into JS instead, and then keep everything as is

```
const map = {md:COMP_SPINNER_SIZE_MD, lg:COMP_SPINNER_SIZE_LG, sm:COMP_SPINNER_SIZE_SM'}

 <div style={{
 '--g-size':{map[size]}
 }} />
    
```

Reply:
THanks @rlingineni ! I went with option 2. 👍